### PR TITLE
feat: make some ffmpeg arguments configurable via env variables

### PR DESCRIPTION
**Pull request by ribeaud**
Introduce two new environment variables:
- `JIBRI_RECORDING_VIDEO_ENCODE_PRESET`
- `JIBRI_RECORDING_CONSTANT_RATE_FACTOR`

### CODE REVIEW

File: jibri/rootfs/defaults/jibri.conf [hunk 1]
```diff
@@ -118,6 +123,20 @@ jibri {
...
       audio-device =  "default"
+      // Available presets: ultrafast, superfast, veryfast, faster, fast, medium,
+      // slow, slower, veryslow, placebo
+      video-encode-preset = "{{ $JIBRI_RECORDING_VIDEO_ENCODE_PRESET }}"
+      // The range of the CRF scale is 0-51, where 0 is lossless,
+      // 23 is the default, and 51 is worst quality possible.
+      h264-constant-rate-factor = "{{ $JIBRI_RECORDING_CONSTANT_RATE_FACTOR }}"
+      // Settings based upon the specifics of your input. Current tunings include:
+      // film, animation, grain, stillimage, fastdecode, zerolatency, psnr, ssim
+      tune = "{{ $JIBRI_RECORDING_TUNE }}"
```
Comment (saghul) [inline jibri/rootfs/defaults/jibri.conf:134]: I don't see this being configurable.

File: jibri/rootfs/defaults/jibri.conf [hunk 2]
```diff
@@ -118,6 +123,20 @@ jibri {
       audio-source = "pulse"
       // The audio device that will be used to capture audio on Linux
       audio-device =  "default"
+      // Available presets: ultrafast, superfast, veryfast, faster, fast, medium,
+      // slow, slower, veryslow, placebo
+      video-encode-preset = "{{ $JIBRI_RECORDING_VIDEO_ENCODE_PRESET }}"
+      // The range of the CRF scale is 0-51, where 0 is lossless,
+      // 23 is the default, and 51 is worst quality possible.
+      h264-constant-rate-factor = "{{ $JIBRI_RECORDING_CONSTANT_RATE_FACTOR }}"
```
Comment (saghul) [inline jibri/rootfs/defaults/jibri.conf:131]: No quotes needed on this one.

File: jibri/rootfs/defaults/jibri.conf [hunk 3]
```diff
@@ -118,6 +123,20 @@ jibri {
...
+      // Settings based upon the specifics of your input. Current tunings include:
+      // film, animation, grain, stillimage, fastdecode, zerolatency, psnr, ssim
+      tune = "{{ $JIBRI_RECORDING_TUNE }}"
+      // Limits the output to a specific H.264 profile. List of profiles that x264 supports:
+      // baseline, main, high, high10, high422, high444
+      profile:v = "{{ $JIBRI_RECORDING_PROFILE }}"
```
Comment (saghul) [inline jibri/rootfs/defaults/jibri.conf:137]: I don't see this being configurable.

File: jibri/rootfs/defaults/jibri.conf [hunk 4]
```diff
@@ -118,6 +123,20 @@ jibri {
...
+      // Set of constraints on the encoding parameters, defined in Annex A of the H.264 standard
+      level = "{{ $JIBRI_RECORDING_LEVEL }}"
```
Comment (saghul) [inline jibri/rootfs/defaults/jibri.conf:139]: Ditto.

**Review by saghul — CHANGES_REQUESTED** [review]

File: jibri/rootfs/defaults/jibri.conf (hunk 1 above)
Comment (ribeaud) [reply jibri/rootfs/defaults/jibri.conf:134]: Are you saying that I can only consider options defined in [this](https://github.com/jitsi/jibri/blob/master/src/main/kotlin/org/jitsi/jibri/capture/ffmpeg/FfmpegCapturer.kt) file?

Or are you saying that _tune_ is not configurable in **ffmpeg** (which is wrong IMHO and up the [documentation](https://trac.ffmpeg.org/wiki/Encode/H.264#a2.Chooseapresetandtune))?

File: jibri/rootfs/defaults/jibri.conf [hunk 5]
```diff
@@ -2,6 +2,11 @@
 {{ $JIBRI_RECORDER_USER := .Env.JIBRI_RECORDER_USER | default "recorder" -}}
 {{ $JIBRI_USAGE_TIMEOUT := .Env.JIBRI_USAGE_TIMEOUT | default "0" -}}
 {{ $JIBRI_RECORDING_RESOLUTION := .Env.JIBRI_RECORDING_RESOLUTION | default "1280x720" -}}
+{{ $JIBRI_RECORDING_VIDEO_ENCODE_PRESET := .Env.JIBRI_RECORDING_VIDEO_ENCODE_PRESET | default "veryfast" -}}
+{{ $JIBRI_RECORDING_CONSTANT_RATE_FACTOR := .Env.JIBRI_RECORDING_CONSTANT_RATE_FACTOR | default "25" -}}
```
Comment (ribeaud) [inline jibri/rootfs/defaults/jibri.conf:6]: May I remove the quotes here as well for the default? So `default 25` instead of `default "25"`.

Comment (saghul) [reply jibri/rootfs/defaults/jibri.conf:6]: Yep, that should do.

File: jibri/rootfs/defaults/jibri.conf (hunk 1 above)
Comment (saghul) [reply jibri/rootfs/defaults/jibri.conf:134]: The former.

Comment (ribeaud) [reply jibri/rootfs/defaults/jibri.conf]: I replaced the non-configurable ones with `JIBRI_RECORDING_FRAMERATE`, `JIBRI_RECORDING_QUEUE_SIZE` and `JIBRI_RECORDING_STREAMING_MAX_BITRATE`.

**Review by saghul — APPROVED** [review]

**Comment by ribeaud** [comment]
> Cheers!
🍾

**Pull request merged**